sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -132,6 +136,12 @@ public static ClientPipeline Create(
         // Add retry policy.
         policies[index++] = options.RetryPolicy ?? ClientRetryPolicy.Default;
 
+        if (policies[index] is ClientRetryPolicy retryPolicy)
+        {
+            ILogger logger = options.LoggingOptions.LoggerFactory.CreateLogger<ClientRetryPolicy>();
+            retryPolicy.LogHandler = new LoggingHandler(logger, new PipelineMessageSanitizer(options.LoggingOptions.AllowedQueryParameters.ToArray(), options.LoggingOptions.AllowedHeaderNames.ToArray()));
```

Comment:
Why is logging managed in the retry policy?

Reply:
This was my interpretation of the [recommended pattern](https://learn.microsoft.com/en-us/dotnet/core/extensions/logging-library-authors#when-to-use-the-iloggerfactory-interface):
> When you need a logging object that can be passed along to multiple classes so that all of them can emit logs, use ILoggerFactory. It's recommended that each class creates logs with a separate category, named the same as the class. To do this, you need the factory to create unique ILogger<TCategoryName> objects for each class that emits logs. 

I'm not attached to this particular way of implementing it, but the idea is that every class or policy that logs _should_ log to its own ILogger instance.

Reply:
That's interesting.  My mental model was that we would have a single logging component for all of SCM that all SCM pipeline types would log to.  Then, the instance of that component would be held in an internal instance on the pipeline that policies could reference when logging events.

The guidance you linked raises the question -- should SCM have ILoggers for each different class that logs events?  What is the advantage of that approach vs. a single ILogger instance for all of SCM?  Given our other feature goals -- consistency with Azure clients, the public Options APIs we're considering, etc. -- would one approach be preferred to another?  If we later wanted to support SCM-based client logging using an approach similar to the SCM approach, would we want to do one or the other?  (i.e. should OpenAI's ChatClient write to a different ILogger than its FineTuningClient, e.g.?)  Then, if there is a single logging abstraction type in SCM, would it manage the different ILogger instances for different classes?  Does this guidance break that abstraction, and if it does, do the benefits outweigh the costs of that?

Reply:
It might be worth looking at Aspire needs for the OAI client and logging as an interesting customer-scenario focused lens on these questions.

Reply:
I think SCM should have ILoggers for every class

Reply:
I thought about creating abstractions for each class, and each abstraction holds the set of logs that class needs. For example, MessageLoggingPolicyLoggingHandler would hold HTTP message logs, TransportLoggingHandler would hold transport logs, etc

Reply:
We could start with just one though, since adding more fine-tuned filtering categories later on wouldn't be a breaking change.